src/runtime_src/core/pcie/tools/xbmgmt/cmd_scan.cpp

```diff
@@ -28,20 +28,30 @@ int scanHandler(int argc, char *argv[])
     if (argc != 1)
         return -EINVAL;
 
-    size_t total = pcidev::get_dev_total(false);
-    
-    if (total == 0) {
-        std::cout << "No card is found!" << std::endl;
-    } else {
-        for (size_t i = 0; i < total; i++) {
-            auto dev = pcidev::get_dev(i, false);
-            if (dev->is_ready)
-                std::cout << " ";
-            else
-                std::cout << "*";
-            std::cout << dev << std::endl;
+    try {
+        size_t total = pcidev::get_dev_total(false);
+
+        if (total == 0) {
+            std::cout << "No card is found!" << std::endl;
+        } else {
+            for (size_t i = 0; i < total; i++) {
+                auto dev = pcidev::get_dev(i, false);
+                if (dev->is_ready)
```

Comment:
You might wish to replace the next 5 lines of code with the following:
`std::cout << (dev->is_ready) ? " " : "*" << dev << std::endl;`